pymmcore_plus/core/_mmcore_plus.py

```diff
@@ -126,15 +126,36 @@ def setDeviceAdapterSearchPaths(self, adapter_paths: ListOrTuple[str]) -> None:
         logger.info(f"setting adapter search paths: {adapter_paths}")
         super().setDeviceAdapterSearchPaths(adapter_paths)
 
-    def loadSystemConfiguration(self, fileName="demo") -> None:
-        if fileName.lower() == "demo":
-            if not self._mm_path:
-                raise ValueError(  # pragma: no cover
-                    "No micro-manager path provided. Cannot load 'demo' file.\nTry "
-                    "installing micro-manager with `python install_mm.py`"
-                )
-            fileName = (Path(self._mm_path) / "MMConfig_demo.cfg").resolve()
-        super().loadSystemConfiguration(str(fileName))
+    def loadSystemConfiguration(
+        self, fileName: str | Path = "MMConfig_demo.cfg"
+    ) -> None:
+        """Load a config file.
+
+        Will first search for *fileName* in the following order:
+        1. Relative to the current working directory
+        2. In the micromanager install folder
+        3. As an absolute path
+        """

```

Comment:
i feel like relative and absolute paths don't warrant separate treatment... that should be in the path itself (i.e. if it starts with `/` or not)...

and the code itself could be simplified significantly to something like:

```py
fpath = Path(fileName).expanduser()
if not fpath.exists() and not fpath.is_absolute() and self._mm_path:
    fpath = Path(self._mm_path) / fileName
if not fpath.exists():
    raise FileNotFoundError(f"Path does not exist: {fpath}")
return super().loadSystemConfiguration(str(fpath.resolve()))
```

Reply:
I think the issue is that I didn't fully understand what `.resolve()` does. This is much better - will modify

Reply:
Good mockup - seems like it just works out of the box